File: Solutions/Solution58.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int lengthOfLast = 0;

        int endIndex = s.size() - 1;
        while (s[endIndex] == ' ')
        {
            endIndex--;
        }

        while (endIndex >= 0 && s[endIndex] != ' ')
        {
            lengthOfLast++;
            endIndex--;
        }

        return lengthOfLast;
    }
};

File: Solutions/Solution58.java
class Solution {
    public int lengthOfLastWord(String s) {
        int endIndex = s.length() - 1;

        char[] words = s.toCharArray();

        while (words[endIndex] == ' ') {
            endIndex--;
        }

        int lengthWord = 0;
        while (endIndex >= 0 && words[endIndex] != ' ') {
            lengthWord++;
            endIndex--;
        }

        return lengthWord;
    }
}
